### PR TITLE
Add funnel for logging retention

### PR DESCRIPTION
**Pull request by coreyfloyd**
https://phabricator.wikimedia.org/T126693

### CODE REVIEW

File: Wikipedia/Code/NSUserDefaults+WMFExtensions.swift [hunk 1]
```diff
@@ -79,5 +79,35 @@ extension NSUserDefaults {
...
+        let previous = self.wmf_appInstallDate()
+        
+        if previous == nil {
+            self.setObject(date, forKey: "AppInstallDate")
+            self.synchronize()
+        }
+    }
+    
+    public func wmf_appInstallDate() -> NSDate? {
+        if let date = self.objectForKey("AppInstallDate") as? NSDate {
```
Comment (bgerstle) [inline Wikipedia/Code/NSUserDefaults+WMFExtensions.swift]: FYI: you can return the result of the optional cast directly

File: Wikipedia/Code/NSUserDefaults+WMFExtensions.swift [hunk 2]
```diff
@@ -79,5 +79,35 @@ extension NSUserDefaults {
...
+        }
+    }
+
+    public func wmf_setDateLastDailyLoggingStatsSent(date: NSDate) {
+        self.setObject(date, forKey: "DailyLoggingStatsDate")
+        self.synchronize()
+    }
+
+    public func wmf_dateLastDailyLoggingStatsSent() -> NSDate? {
+        if let date = self.objectForKey("DailyLoggingStatsDate") as? NSDate {
```
Comment (bgerstle) [inline Wikipedia/Code/NSUserDefaults+WMFExtensions.swift]: same here

File: Wikipedia/Code/NSUserDefaults+WMFExtensions.swift (hunk 1 above)
Comment (coreyfloyd) [inline Wikipedia/Code/NSUserDefaults+WMFExtensions.swift]: @bgerstle I did that at first and it complained - it wanted me to wrap everything in a force unwrap

Comment (bgerstle) [inline Wikipedia/Code/NSUserDefaults+WMFExtensions.swift]: strange, seems to work in REPL 😕

``` swift
extension NSUserDefaults {
    func foo() -> NSDate? {
        return objectForKey("Foo") as? NSDate
    }
}
NSUserDefaults.standardUserDefaults().foo()
$R1: NSDate? = nil
```

**Pull request merged**